src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -662,6 +673,17 @@ let pp_model ppf ({Program.prog_name; _} as p) =
   pf ppf "@ @[<v 1>@ private:@ @[<v 1> %a@]@ " pp_model_private p ;
   pf ppf "@ public:@ @[<v 1> ~%s() { }" p.prog_name ;
   pf ppf "@ @ std::string model_name() const { return \"%s\"; }" prog_name ;
+  pf ppf
+    {|
+
+  std::vector<std::string> model_compile_info() const {
+    std::vector<std::string> stanc_info;
+    stanc_info.push_back("stanc_version = %s");
+    stanc_info.push_back("stancflags = %s");
+    return stanc_info;
+  }
+  |}
+    "%%NAME%%3 %%VERSION%%" stanc_args_to_print ;
```

Comment:
What happens if you put these two in the string literal itself? e.g. 

`    stanc_info.push_back("stanc_version = %%NAME%%3 %%VERSION%%");`

I would have thought that would work but maybe it makes it harder to read or something?

Reply:
Doh, yes. Thanks!

Reply:
Do you know if this %%VERSION%% names work with stancjs? I tried it once but failed. Or is this only for building the stanc binary?

Reply:
Ah! If it's not working, I probably have a bug in the stancjs deployment code in Jenkins. The way the %%VERSION%% stuff works is very simple, you just run `dune promote` before building and it does the appropriate string substitutions.

Reply:
Ah ok, works locally now! Thanks!! And there is a `dune subst` in the Jenkinsfile so it should also work for the stuff built on Jenkins. Want to add args to stancjs for https://github.com/stan-dev/stanc3/issues/577 and rstan stuff in general and started with --version and bumped into that :)